music/src/core/data.ts

```diff
@@ -396,6 +409,143 @@ export class MelodyConverter extends DataConverter {
   }
 }
 
+/**
+ * Abstract `MelodyControlConverter` class for handling melody control signals.
+ * The `toTensor` method first extracts a monophonic melody from the
+ * `NoteSequence` then extracts the control signal from the melody. The
+ * `toNoteSequence` method is left undefined here but can be used by subclasses
+ * for debugging purposes.
+ *
+ * @param numSteps The length of each sequence.
+ * @param minPitch The minimum melody pitch.
+ * @param maxPitch The maximum melody pitch.
+ * @param ignorePolpyhony Whether to raise an error if multiple notes start at
+ * the same step.
+ * @param numSegments (Optional) The number of conductor segments.
+ */
+abstract class MelodyControlConverter extends DataConverter {
+  readonly minPitch: number;
+  readonly maxPitch: number;
+  readonly ignorePolyphony: boolean;
+  readonly melodyControl: MelodyControl;
+  readonly depth: number;
+  readonly endTensor: tf.Tensor1D;
+
+  constructor(args: MelodyConverterArgs, melodyControl: MelodyControl) {
+    super(args);
+    this.minPitch = args.minPitch;
+    this.maxPitch = args.maxPitch;
+    this.ignorePolyphony = args.ignorePolyphony;
+    this.melodyControl = melodyControl;
+    this.depth = melodyControl.depth;
+  }
+
+  toTensor(noteSequence: INoteSequence): tf.Tensor2D {
+    const melody = Melody.fromNoteSequence(
+        noteSequence, this.minPitch, this.maxPitch, this.ignorePolyphony,
+        this.numSteps);
+    return this.melodyControl.extract(melody);
+  }
+}
+
+/**
+ * Converts between a monophonic, quantized `NoteSequence` containing a melody
+ * and a `Tensor` representing only the *rhythm* of the melody.
+ *
+ * The rhythm is represented as a `numSteps`-by-1 `Tensor` with 1 in the
```

Comment:
I think "`[num_steps, 1]`-shaped `Tensor`" is more readable.

Reply:
Done.